templates/worker/01-worker-kubelet/on-prem/units/kubelet.service.yaml

```diff
@@ -10,6 +10,7 @@ contents: |
 
   [Service]
   Type=notify
+  NotifyAccess=all
```

Comment:
How is this one related?

Reply:
Oh, this is a cool story.

TL;DR="Because we're wrapping kubelet in kubensenter and kubelet is not the direct child of systemd, we need this so kubelet can tell systemd it's running via the `Type=notify` mechanism"

So...

If you see the history of this PR, you can see that we were very consistently failing CI all the way along until the end, and it was very mysterious. Running clusters with the same code, when doing a manual inspection everything looked healthy; `oc get co` showed everything available, `oc get nodes` showed everything Ready.  So the weird CI failures made no sense. For example: CI failed running 'rsh' to cat a file. Manual execution of the same operation always succeeded.  And manually running the MCO e2e tests sometimes passed, or failed in other weird ways intermittently.

The main point of this change is that we want to run `kubelet` inside of the new `kubensenter` wrapper, and due to what turns out to be a short-sighted design choice, that `kubensenter` wrapper does not `exec`, but keeps itself in the PS tree, as an audit that it's there. An important and initially-thought benign impact is that while `kubelet` used to be the direct parent of systemd in the process tree, it is now the child of `kubensenter`.

Now we go deeper into systemd. It turns out that for a `Type=notify` service like this one, unless you say otherwise this `NotifyAccess` defaults to the value `main`, which means that systemd will actively ignore any notification signal that is NOT from the PID directly started by Systemd's ExecStart stanza. Because of this, systemd would start kubelet, let it run for a while, then timeout waiting for the notification signal and restart it.

These periodic restarts would correspond to random CI failures, and were largely invisible to manual checks, especially because when `oc get co` returns everything "available" there's usually no reason to check whether kubelet itself is healthy!  But it turns out, it wasn't...

And now you know the rest of the story.

Reply:
I have a pending change in kubelet to address this: https://github.com/openshift/kubernetes/pull/1350

Once that's in and kubensenter calls `exec`, we can either edit this PR to remove this `NotifyAccess=all` line, or if this is already merged, I'll launch a follow-up PR to clean this up.

Up to you if you'd prefer to hold this PR on that change, or let this in as-is and fix it later.

Reply:
I think we should make the change to have kubensenter call exec and drop this line.

Reply:
I have removed these `NotifyAccess=all` lines.  Once the right kubelet change makes its way through the CI lanes here will pass again and this can be merged :)